### ISSUE TITLE
Cascade Delete

### ISSUE THREAD

**Issue by mbrahimi617** [redux-orm/redux-orm#423]
Is there a way to setup Redux-ORM such that related models cascade delete by default?

**Comment by haveyaseen** [comment]
Right now the delete method mimics SQL's `ON DELETE SET NULL` instead of `ON DELETE CASCADE`.

If you take a look at this block of code it becomes clear what you need to do.

https://github.com/redux-orm/redux-orm/blob/3eccce7342c194d2d871761637ad4f2a9deaf6dd/src/Model.js#L758-L779

I haven't tested this but in theory you could extend the Model class and overwrite the `delete` method (or just copy it right over and overwrite it in your own custom Model class).

```js
import { Model } from "redux-orm";
class CascadingDeleteModel extends Model {
    delete() {
        const { virtualFields } = this.getClass();
        // eslint-disable-next-line guard-for-in, no-restricted-syntax
        for (const key in virtualFields) {
            const field = virtualFields[key];
            if (field instanceof ManyToMany) {
                // Delete any many-to-many rows the entity is included in.
                this[key].delete();
            } else if (field instanceof ForeignKey) {
                const relatedQs = this[key];
                if (relatedQs.exists()) {
                    relatedQs.delete();
                }
            } else if (field instanceof OneToOne) {
                // Set null to any foreign keys or one to ones pointed to
                // this instance.
                if (this[key] !== null) {
                    this[key].delete();
                }
            }
        }
        super.delete();
    }
}
```

**Comment by mbrahimi617** [comment]
It seems incredibly verbose and low level, and requires a lot of deciphering the source. If the object of the library is to simulate a SQL database on the frontend as a Redux store, then it should have some of these standard DELETE options built-in in my opinion.

**Comment by haveyaseen** [comment]
You're right, this could be part of Redux-ORM. However I don't know what a good API would look like. Adding to the API surface area is always a dangerous move and it should have some foresight. Open to suggestions.

**Comment by mbrahimi617** [comment]
I would probably accept another argument in the relationship where you define your relationship fields just as you would in any other ORM, backend or frontend. Personally, I come from a Django background in which the ForeignKey field accepts another model to which it refers. It optionally accepts the name of the reverse relationship, and an `on_delete` constant such as `CASCADE`, `SET_NULL`, etc.

https://stackoverflow.com/questions/38388423/what-does-on-delete-do-on-django-models

As it now stands, the API already supports the first two arguments. Obviously either said than done, because I certainly can't decipher the source and submit a PR, but you'd have to add that last argument for delete events. Not a related issue, but you should also decouple models from ORM instances entirely such that they can be handpicked and reused in different instances as mentioned in #417 and #422.

**Comment by mbrahimi617** [comment]
Any updates on this?
